libcudacxx/.upstream-tests/test/support/iterator_traits_cpp17_iterators.h

```diff
@@ -0,0 +1,120 @@
+//===----------------------------------------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+// SPDX-FileCopyrightText: Copyright (c) 2023 NVIDIA CORPORATION & AFFILIATES.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef TEST_SUPPORT_ITERATOR_TRAITS_ITERATOR_TRAITS_CPP17_ITERATORS
```

Comment:
Note that proper test iterators are added later in https://github.com/NVIDIA/cccl/pull/128/commits/d43c6ac7825674a00af7e1857fecb58f8bc70b0e